server.js

```diff
@@ -1,17 +1,73 @@
-// Require the framework and instantiate it
 const fastify = require('fastify')({ logger: true })
+const { db_connect, db_setup, new_print_job, new_workflow, new_workflow_step, new_simulation_report } = require("./mongodb.js");
 
-// Declare a route
-fastify.get('/', function handler (request, reply) {
-  reply
-        .code(200)
-        .send('hello client!');
-});
+//TODO: can this value be tracked somwhere else so we don't have to remember to change it before merging?
+const port_num = 80; //Default: 80
 
-// Run the server!
-fastify.listen({ port: 80, host: '0.0.0.0' }, (err, address) => {
-  if (err) {
+
+async function start(){ 
+  try { 
+    [client, database] = await db_connect();
+    await db_setup(database);
+    
+    setup_gets(database);
+    setup_posts(database);
+    
+    fastify.listen({ port: port_num, host: '0.0.0.0' }, (err, address) => {
+      if (err) throw err;
+    });
+
+  } catch (err){
     fastify.log.error(err);
     process.exit(1);
   }
-});
\ No newline at end of file
+}
+
+
+function setup_gets(database){
+  fastify.get('/', function handler (request, reply) {
+    reply.code(200).send('hello client!');
+  });
+}
+
+
+function setup_posts(database){
+  //TODO: Test how mongoDB handles multiple simultaneous requests
+  fastify.post('/createJob', async (request, reply) => {
+    await fastify_post_helper(reply, database, new_print_job, 
+      [request.body.Title, request.body.PageCount, request.body.RasterizationProfile]);
+  });
+
+  fastify.post('/createWorkflow', async (request, reply) => {
+    await fastify_post_helper(reply, database, new_workflow, 
+      [request.body.Title, request.body.WorkflowSteps]);
+  });
+
+  fastify.post('/createWorkflowStep', async (request, reply) => {
+    await fastify_post_helper(reply, database, new_workflow_step, 
+      [request.body.Title, request.body.PreviousStep, request.body.NextStep, request.body.SetupTime, request.body.TimePerPage]);
+  });
+
+  //TODO: could the helper function be modified to support this?
+  //TODO: reformat to a get instead of post?
+  fastify.post('/query', async (request, reply) => {
+    message = ""
+    code = 200;
+    collection = database.collection(request.body.CollectionName);
+    try { message = await collection.find(request.body.Query).toArray(); }
+    catch(err) {message = err; code = 500; }
+    reply.code(code).send(message);
+  });
+}
+
+
+async function fastify_post_helper(reply, database, func, args){
```

Comment:
I know you want to implement DRY here but eventually, errors are going to be more robust than `500`. I think you should just include this functionality in `/createJob`, `/createWorkflow`, and `/createWorkflowStep`.